src/lib/dexrt.cpp

```diff
@@ -138,7 +140,34 @@ double randunif(uint64_t keypair) {
   return out - 1;
 }
 
-void showFloat(char **resultPtr, float x) {
+void showInt32(char **resultPtr, int32_t x) {
+  auto p = reinterpret_cast<char*>(malloc_dex(100));
+  auto n = sprintf(p, "%" PRId32, x);
+  auto result1Ptr = reinterpret_cast<int32_t*>(resultPtr[0]);
+  auto result2Ptr = reinterpret_cast<char**>(  resultPtr[1]);
+  *result1Ptr = n;
+  *result2Ptr = p;
+}
+
+void showInt64(char **resultPtr, int64_t x) {
+  auto p = reinterpret_cast<char*>(malloc_dex(100));
+  auto n = sprintf(p, "%" PRId64, x);
+  auto result1Ptr = reinterpret_cast<int32_t*>(resultPtr[0]);
+  auto result2Ptr = reinterpret_cast<char**>(  resultPtr[1]);
+  *result1Ptr = n;
+  *result2Ptr = p;
+}
+
+void showFloat32(char **resultPtr, float x) {
+  auto p = reinterpret_cast<char*>(malloc_dex(100));
+  auto n = sprintf(p, "%.4f", x);
```

Comment:
maybe let's not restrict the expansion here?

Reply:
Changed format string from `"%.4f"` to `"%f"` for `float` and `double`. `make tests` still passes.

Reply:
I still need to add the newly introduced `prelude-tests.dx` as a testcase in `makefile`.

Reply:
Done. I used the format strings explained in [this Stack Overflow answer](https://stackoverflow.com/a/19897395/5365899):

```cpp
// float, __FLT_DECIMAL_DIG__ = 9 for IEEE float
snprintf(buffer, STRING_BUFFER_SIZE, "%.*g", __FLT_DECIMAL_DIG__, x);
// double, __FLT_DECIMAL_DIG__ = 9 for IEEE double
snprintf(buffer, STRING_BUFFER_SIZE, "%.*g", __DBL_DECIMAL_DIG__, x);
```

Related resources: https://en.cppreference.com/w/cpp/header/cfloat

---

The `Show` instance tests behave nicer now, but some `show` results are inaccurate (compared with Python or Haskell or Swift):
```
:p show (123.456789: Float32)
> (AsList 10 ['1', '2', '3', '.', '4', '5', '6', '7', '8', '7'])

:p show (FToF64 123.456789: Float64)
> (AsList 16 [ '1'
> , '2'
> , '3'
> , '.'
> , '4'
> , '5'
> , '6'
> , '7'
> , '8'
> , '7'
> , '1'
> , '0'
> , '9'
> , '3'
> , '7'
> , '5' ])
```

The eventual direction is to rewrite printing (`Show` instances) within Dex itself, à la [`showFloat` from GHC](https://sourcegraph.com/github.com/ghc/ghc@41c64eb5db50c80e110e47b7ab1c1ee18dada46b/-/blob/libraries/base/GHC/Float.hs#L696-704). That should improve the accuracy of the printed representation and is a step towards Dexiness.